runtime/src/main/java/org/corfudb/runtime/collections/StreamManager.java

```diff
@@ -0,0 +1,238 @@
+package org.corfudb.runtime.collections;
+
+import com.google.protobuf.Message;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+
+import org.corfudb.protocols.logprotocol.MultiObjectSMREntry;
+import org.corfudb.protocols.wireprotocol.ILogData;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.exceptions.StreamSubscriptionException;
+import org.corfudb.runtime.view.Address;
+import org.corfudb.runtime.view.ObjectsView;
+import org.corfudb.runtime.view.StreamOptions;
+import org.corfudb.runtime.view.TableRegistry;
+import org.corfudb.runtime.view.stream.IStreamView;
+
+/**
+ * A simple thread based subscription engine where each subscriber or listener gets
+ * a thread which will listen on the tables of interest to it and moves at a rate
+ * which is the same as the consumption.
+ *
+ * Created by hisundar on 04/28/2020.
+ */
+@Slf4j
+public class StreamManager {
+
+    private static final int DEFAULT_NUM_SUBSCRIBERS = 6; // Arbitrary sanity check value.
+    private static final int DEFAULT_SLEEP_MILLIS = 50; // Default time to sleep
+    private final int maxSubscribers;
+
+    /**
+     * Corfu Runtime.
+     */
+    @Getter
+    private final CorfuRuntime runtime;
+
+
+    @Getter
+    ScheduledExecutorService scheduledExecutorService;
+
+    @AllArgsConstructor
+    @Getter
+    class StreamSubscriber {
+        SubscriberTask subscriberTask;
+        ScheduledFuture<?> scheduledFuture;
+    }
+
+    /**
+     * Map of StreamingSubscriptionThreads.
+     * <p>
+     * This is the map of all streaming subscriptions.
+     */
+    @Getter
+    private final Map<StreamListener, StreamSubscriber> subscriptions;
+
+    public StreamManager(@Nonnull CorfuRuntime runtime, int maxSubscribers) {
+        this.runtime = runtime;
+        this.maxSubscribers = maxSubscribers;
+        this.subscriptions = new HashMap<>();
+        this.scheduledExecutorService = Executors.newScheduledThreadPool(maxSubscribers);
```

Comment:
That is what I added before, please see comments from @xnull  and @Maithem . I am open to either suggestion. 
Not having a limit means that it will be up to the caller to ensure that they don't create too many subscribers but if they do know what they are doing, then we don't limit that behavior.